python/pyiceberg/catalog/hive.py

```
@@ -0,0 +1,262 @@
+        identifier: Union[str, Identifier],
+        schema: Schema,
+        location: Optional[str] = None,
+        partition_spec: Optional[PartitionSpec] = None,
+        properties: Optional[Properties] = None,
+    ) -> Table:
+        database_name, table_name = self.identifier_to_tuple(identifier)
+        current_time_millis = int(time())
+        tbl = HiveTable(
+            dbName=database_name,
+            tableName=table_name,
+            owner=getuser(),
+            createTime=current_time_millis // 1000,
+            lastAccessTime=current_time_millis // 1000,
+            sd=self._storage_descriptor(schema, location),
+            tableType="EXTERNAL_TABLE",
+            parameters={"EXTERNAL": "TRUE"},
+        )
+        try:
+            with self._client as open_client:
+                open_client.create_table(tbl)
+                hive_table = open_client.get_table(dbname=database_name, tbl_name=table_name)
+        except AlreadyExistsException as e:
+            raise AlreadyExistsError(f"Table {database_name}.{table_name} already exists") from e
+        return self._convert_hive_into_iceberg(hive_table)
+
+    def load_table(self, identifier: Union[str, Identifier]) -> Table:
+        database_name, table_name = self.identifier_to_tuple(identifier)
+        try:
+            with self._client as open_client:
+                hive_table = open_client.get_table(dbname=database_name, tbl_name=table_name)
+        except NoSuchObjectException as e:
+            raise NoSuchTableError(f"Table {database_name}.{table_name} not found") from e
+
+        return self._convert_hive_into_iceberg(hive_table)
+
+    def _drop_table(self, identifier: Union[str, Identifier], delete_data: bool = False) -> None:
+        database_name, table_name = self.identifier_to_tuple(identifier)
+        try:
+            with self._client as open_client:
+                open_client.drop_table(dbname=database_name, name=table_name, deleteData=delete_data)
```

I don't think that we want the Hive Metastore to delete data. When purging a table, we need to only delete the files that are owned by the table and Hive has no concept of that. It's unlikely that the caller knows much about it, too. Right now in the Java implementation, the client will traverse the reachability set and drop data files. That's what we should eventually do here.

For now, we can just throw an exception in `purge_table`.

---

Got it, updated it, and added a comment